src/DataStructures/Variables.hpp

```diff
@@ -829,6 +829,16 @@ bool operator!=(const Variables<TagsList>& lhs,
   return not(lhs == rhs);
 }
 
+template <typename... TagsLhs, typename... TagsRhs,
+          Requires<not std::is_same<tmpl::list<TagsLhs...>,
```

Comment:
We've had weird compiler errors with some GCC versions when using that with variables so that's why I'm avoiding it. I don't know if it'll be an issue, but it was rather annoying to have the random failures.

Reply:
IIRC the gcc ICEs in the variables code were coming from `flat_all_v`? Were there also issues from `is_same_v` that I've since forgotten?

Reply:
I thought there used to be some related to this, but maybe I'm confusing the two :man_shrugging:

Reply:
I think they were both happening.  GCC seems to generally hate this file.